### DISCUSSION TITLE
[Feature Request] Misc. Thaumcraft improvements

### DISCUSSION THREAD

**Discussion by Vapaman** [ACGaming/UniversalTweaks#341] [Bugfix Suggestions]
Hello!

The new thaumcraft additions this already brings are a nice touch :D
So here is some additions and improvements that would be nice to see implemented:

**Old version visual parity:**

- Silverwood leaves don't produce particles like they do in older thaumcraft versions. (Might be intentional to reduce lag)

- The overlay when a warp effect occurs is not implimented/does not play. - I don't know wheter or not this is intentional.

- Hungry chest nomming animation doesn't play

**Sound Fixes:**
- Runic shielding is supposed to make a sound when charged 
https://github.com/Azanor/thaumcraft-beta/issues/1792

- There seems to be a sound that is supposed to play when failing to create a focus. Currently it only plays it in multiplayer for other nearby players that are not using the manipulator.

- The creeper warp effect is buggy as it plays for other nearby players, not the warped player themself.

- The arcane bore is supposed to play a sound when active, but it only does it occasionally for a split of a second.

**Misc.:**
- Potions from pech trading are messed up. Azanor didn't seen to change the potion IDs from their 1.8 state when porting. This is already fixed by the mod **[ThaumTweaks](https://github.com/GrigLog/ThaumTweaks)** but it the mod seems to be pretty buggy and filled with other features, which makes it less than ideal for modpack use.

- During the rare warp event that removes warp ("You have a moment of clarity." ), _the warp event_ text and _the warp removed_ text are overlayed, making it unreadable.

- Shift-clicking things into the research table does not work

- Some research cards don't seem to show up e.g. Balance (mentioned in the Thaumonomicon) and Analyze

- Editing an already made foci is janky: when you exit the manipulator it removes the pattern.

- When remaking a foci, exiting and then re-entering the manipulator gui messes up the focus and makes it unusable.

**Comment by Turkey9002** [comment]
Also would like to mention the infamous experience level bug fixed by [Thaumic Fixes](https://www.curseforge.com/minecraft/mc-mods/thaumic-fixes) since it doesn't look like that mod is maintained anymore and that's the only bug it fixes. There's also a fix for aspect registrations all the way down on this issue page [here](https://github.com/Azanor/thaumcraft-beta/issues/1619).

**Comment by Vapaman** [comment]
Oh that aspect bug is a real annoyance. Also totally forgot about the experience bug.

**Comment by Krutoy242** [comment]
Since of this bugs was fixed by mod Sledgehammer
